app/src/main/java/org/simple/clinic/drugs/EditMedicinesUiRenderer.kt

```diff
@@ -1,9 +1,42 @@
 package org.simple.clinic.drugs
 
 import org.simple.clinic.drugs.selection.PrescribedDrugUi
+import org.simple.clinic.drugs.selection.ProtocolDrugListItem
+import org.simple.clinic.drugs.selection.entry.CustomPrescribedDrugListItem
 import org.simple.clinic.mobius.ViewRenderer
+import org.simple.clinic.protocol.ProtocolDrugAndDosages
 
 class EditMedicinesUiRenderer(private val ui: PrescribedDrugUi) : ViewRenderer<EditMedicinesModel> {
   override fun render(model: EditMedicinesModel) {
+    if (model.prescribedDrugs == null || model.protocolDrugs == null)
+      return
+
+    if (model.prescribedDrugs != null && model.protocolDrugs != null)
+      renderPrescribedProtocolDrugs(model.prescribedDrugs, model.protocolDrugs)
```

Comment:
Suggestion: if we move the conditional for checking whether a model is ready to render to the model, we can simplify this a lot.

```kotlin
class EditMedicinesModel {
  val hasLoadedPrescriptionsAndProtocols: Boolean
    get() = prescribedDrugs != null && protocolDrugs != null
}

class EditMedicinesUiRenderer {
  fun render() {
    if(model.hasLoadedPrescriptionsAndProtocols) {
      // render UI here
    }
  }
}
```

Reply:
I just realized that I don't need the first `if` condition. I will remove that as well.

Reply:
Also, if I move the condition to the model, I will have to add null safety when accessing the two fields instead of being able to use Kotlin's smart cast.
I think it's better this way. Please suggest.

Reply:
I personally prefer extracting boolean logic out into a property or a local variable and give it a name that has some meaning, so I suggested this.

Even if we don't move this to the model, I think we can also remove that early return since it doesn't add any value since we have another check to render the UI in the first place.